codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -66,412 +72,1112 @@
+                    String msgString = frame.toString();
+                    ReferenceCountUtil.release(frame);
+                    throw new IllegalArgumentException(
+                            "Stream " + frame.stream() + " must not be set on the frame: " + msgString);
+                }
+                return frame;
+            }
+
+            private ChannelFuture write0(Object msg) {
+                ChannelPromise promise = ctx.newPromise();
+                Http2MultiplexCodec.this.write(ctx, msg, promise);
+                return promise;
+            }
+
+            @Override
+            public void flush() {
+                if (flush) {
+                    // If we are current channelReadComplete(...) call we should just mark this Channel with a flush
+                    // pending. We will ensure we trigger ctx.flush() after we processed all Channels later on and so
+                    // aggregate the flushes. This is done as ctx.flush() is expensive when as it may trigger an
+                    // write(...) or writev(...) operation on the socket.
+                    if (inFireChannelReadComplete) {
+                        flushPending = true;
+                    } else {
+                        flush0(ctx);
+                    }
+                } else {
+                    // There is nothing to flush so this is a NOOP.
+                }
+            }
+
+            @Override
+            public ChannelPromise voidPromise() {
+                return unsafeVoidPromise;
+            }
+
+            @Override
+            public ChannelOutboundBuffer outboundBuffer() {
+                // Always return null as we not use the ChannelOutboundBuffer and not even support it.
+                return null;
+            }
+        }
+
+        /**
+         * {@link ChannelConfig} so that the high and low writebuffer watermarks can reflect the outbound flow control
+         * window, without having to create a new {@link WriteBufferWaterMark} object whenever the flow control window
+         * changes.
+         */
+        private final class Http2StreamChannelConfig extends DefaultChannelConfig {
+
+            Http2StreamChannelConfig(Channel channel) {
+                super(channel);
+                setRecvByteBufAllocator(Http2StreamChannelRecvByteBufAllocator.INSTANCE);
+            }
+
+            /**
+             * @deprecated  Use {@link #getWriteBufferWaterMark()} instead.
+             */
+            @Override
+            @Deprecated
+            public int getWriteBufferHighWaterMark() {
+                return max(0, (int) min(Integer.MAX_VALUE, outboundFlowControlWindow));
+            }
+
+            /**
+             * @deprecated  Use {@link #getWriteBufferWaterMark()} instead.
+             */
+            @Override
+            @Deprecated
+            public int getWriteBufferLowWaterMark() {
+                return getWriteBufferHighWaterMark();
+            }
+
+            @Override
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the missing @deprecated Javadoc tag. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)